go-controller/pkg/node/gateway_iptables.go

```diff
@@ -421,7 +435,12 @@ func getGatewayIPTRules(service *kapi.Service, svcHasLocalHostNetEndPnt bool) []
 				// case1 (see function description for details)
 				// Port redirect host -> Nodeport -> host traffic directly to endpoint
 				for _, clusterIP := range clusterIPs {
-					rules = append(rules, getNodePortETPLocalIPTRules(svcPort, clusterIP, int32(svcPort.TargetPort.IntValue()), svcHasLocalHostNetEndPnt, svcTypeIsETPLocal)...)
+					// Adhere to the addressFamily filter if it's 4 or 6.
+					if addressFamily == 0 ||
+						addressFamily == 6 && utilnet.IsIPv6String(clusterIP) ||
```

Comment:
`if addressFamily == 0 || addressFamily == 6 && utilnet.IsIPv6String(clusterIP) || 			addressFamily == 4 && utilnet.IsIPv4String(clusterIP) {`

wait if we are dual stack OR if its a IPV6 singlestack and clusterIP service OR if its a IPV4 singlestack and clusterIP service.... we add the rule to the list of creation or deletion?